src/overlay005/event_check.c

```diff
@@ -0,0 +1,1095 @@
+        if (FieldEventCheck_Sign(fieldSystem) == TRUE) {
+            return TRUE;
+        }
+    }
+
+    if (eventCheck->interact) {
+        int interactCheck;
+        MapObject *object;
+
+        if (sub_02071CB4(fieldSystem, 9) == TRUE) {
+            interactCheck = FieldEventCheck_DistortionInteract(fieldSystem, &object);
+        } else {
+            interactCheck = sub_0203CA40(fieldSystem, &object);
+        }
+
+        if (interactCheck == TRUE) {
+            if (sub_0205F588(fieldSystem->playerAvatar) == TRUE) {
+                sub_0205F5E4(fieldSystem->playerAvatar, PlayerAvatar_GetDir(fieldSystem->playerAvatar));
+            }
+
+            if (sub_02062950(object) != 0x9) {
+                sub_0203E880(fieldSystem, sub_02062960(object), object);
+            } else {
+                sub_0203E880(fieldSystem, 0, object);
+            }
+
+            return TRUE;
+        }
+
+        int distortionCheck = sub_0205F108(fieldSystem->playerAvatar);
+
+        if (distortionCheck == AVATAR_DISTORTION_STATE_NONE || distortionCheck == AVATAR_DISTORTION_STATE_ACTIVE) {
+            int event = sub_0203CA6C(fieldSystem, (void *)sub_0203A440(fieldSystem), sub_0203A448(fieldSystem));
+
+            if (event != 0xffff) {
+                sub_0203E880(fieldSystem, event, NULL);
+                return TRUE;
+            }
+        }
+
+        if (distortionCheck == AVATAR_DISTORTION_STATE_NONE) {
+            int event;
+
+            if (ov5_021EFB40(fieldSystem, &event)) {
+                sub_0203E880(fieldSystem, event, NULL);
+                return TRUE;
+            }
+        }
+
+        int distortionDir = sub_0205EAA0(fieldSystem->playerAvatar);
+        u32 distortionPermissions = sub_020616F0(fieldSystem->playerAvatar, distortionDir);
+        int distortionEvent = FieldEventCheck_TilePermissionsToEvent(fieldSystem, distortionPermissions);
+
+        if (distortionEvent != 0xffff) {
+            sub_0203E880(fieldSystem, distortionEvent, NULL);
+            return TRUE;
+        }
+
+        int playerX, playerZ;
+        int playerDir = PlayerAvatar_GetDir(fieldSystem->playerAvatar);
+
+        FieldEventCheck_NextPos(fieldSystem, &playerX, &playerZ);
+
+        if (ov5_021F8410(fieldSystem, playerX, playerZ, playerDir) == TRUE) {
+            return TRUE;
+        }
+    }
+
+    if (eventCheck->mapTransition) {
+        if (sub_02071CB4(fieldSystem, 9) == TRUE) {
+            ov9_0224A800(fieldSystem, eventCheck->transitionDir);
+        } else if (FieldEventCheck_MapTransition(fieldSystem, eventCheck) == TRUE) {
+            FieldEventCheck_TrySetMapConnection(fieldSystem);
+            return TRUE;
+        }
+    }
+
+    if (eventCheck->registeredItem && sub_02069238(fieldSystem) == TRUE) {
+        return TRUE;
+    }
+
+    if (eventCheck->menu && sub_0203A9C8(fieldSystem) == TRUE) {
+        Sound_PlayEffect(1533);
```

Comment:
Replace with SDAT constant (and rename that constant, if you have a strong name for it).